### ISSUE TITLE
LOG_REMOTE_USER is not implemented

### ISSUE THREAD

**Issue by acwwat** [SAP/cf-nodejs-logging-support#29]
Currently I have a use case of using the cf-nodejs-logging-support module to produce logs to Kibana on CF in my MTA. The application is secured via XSUAA and is using passport JWT strategy for authentication via the @sap/xssec module, which I think (hope?) will include login details for remote_user.

Thinking that, I set the LOG_REMOTE_USER env var to true according to the README, but I am still getting just a dash in the records on Kibana. Looking at this module's source code reveals that the value is hardcoded in config.js:

```
    }, {
        name: "remote_user",
        mandatory: true,
        envVarSwitch: "LOG_REMOTE_USER",
        source: {
            type: "static",
            value: "-"
        }
    }, {
```
So although documented, it really doesn't work. I'd suggest to either implement the feature, or at least correct the README to say the feature doesn't work. If there is other ways to log the user name for E2E trace purposes, please let me know too. Thank you.

**Comment by nicklas-dohrn** [comment]
Hello acwwat,
You are totally right that the current Readme and implementation is inconsistent.
I looked into this Issue and I found only very sparse Information about remote user usage in nodejs 
(some mentioning on  [Stackoverflow about getRemoteUser in express.js](https://stackoverflow.com/questions/19430769/how-to-get-the-remoteuser-for-a-request-with-express-js) from 2013, but this is outdated information as there is no mention about remoteUser or remote_user in express, restify or nodejs.
So at this point I can only guess, that this might be a set header in a given request, and I named it "remote-user" for extraction by our library.
(currently only available at /master on this repo)
Is this the functionality you are seeking, or is there any standard regarding this topic I missed?
Regards,
Nicklas

**Comment by acwwat** [comment]
Hi Nicklas,

Thanks for your response. Yes this is a functionality I am seeking. More generally, I am looking for similar behavior as @sap/logging where when enabled, the logged in user ID is set in the remote-user field. Here are couple code snippets from @sap/logging which I've been examining that gives some implementation details.

In lib/contexts/EntryContexts.js, observe that it returns req.user.id (which is set by the passport JWT strategy via @sap/wssec) if it exists:
```
  get user() {
    const req = this._req;
    return req && req.user && req.user.id;
  }
```
I double checked that req.user and req.user.id are standard in passport.js: http://www.passportjs.org/docs/authenticate/

In lib/formats/CFLog.js, observe that it sets the remote_user property in the log record to req.user.id per the code above:

`    entry['remote_user'] = this._env.logUser ? (entryCtx.user || UNKNOWN_STRING) : 'redacted';
`

Perhaps to enable the LOG_REMOTE_USER feature, we can follow similar logic in config.js, if I am not way off with my understanding. Please take a look and let me know what you think. Thank you.

Regards,
Anthony

**Comment by MANUGUPTASAP** [comment]
Hi @acwwat @nicklas-dohrn 

Do we have any update on this issue ?
I am also looking for similar kind of feature.

Thanks

**Comment by nicklas-dohrn** [comment]
Hello all,
Sorry for me missing to answer here.
the functionality mentioned above is already implemented some time ago.

remote-user is the field where the information is load into, when i last updated the issue.
If you need to overwrite this funcionality, or want to provide different information in the "remote-user" field,
you are able to do it like this:
`
remote_user_information = this._env.logUser ? (entryCtx.user || UNKNOWN_STRING) : 'redacted';
log.info("write different remote user info", {"remote_user": remote_user_information});
`
I leave this one open, if this is not the idea you had in mind.

**Comment by christiand93** [comment]
Hi, 
I have checked if we can provide a suitable solution for this issue. Reading the user ID from `req.user.id` definitely makes sense especially since @sap/application router, passport and CAP support this field. #97 updates the related field configuration and should fix this issue.

**Comment by christiand93** [comment]
This feature is available as of version 6.11.0
